### PR TITLE
Add support to enable ECR image scan

### PR DESCRIPTION
**Pull request by hari2192**
Adding support to enable recently launched ECR image scan.

1. Added optional parameter "scan-on-push"
2. If "scan-on-push" parameter is set, then update scanning configuration for ECR repo.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,6 +37,10 @@ steps:
 
   Name of the ECR repository.
 
+- `scan-on-push` (optional, string)
+
+  Value to enable ECR image scan. Accepted values: true or false.
```
Comment (72636c) [inline README.md]:
```suggestion
  Whether to [automatically scan images](https://docs.aws.amazon.com/AmazonECR/latest/userguide/image-scanning.html#scanning-repository) pushed to the ECR repository for vulnerabilities.
  
  Omitting this option will leave the existing image scanning configuration untouched.
```

File: hooks/pre-command [hunk 2]
```diff
@@ -37,6 +37,14 @@ upsert_ecr() {
   aws ecr put-lifecycle-policy \
   --repository-name "${repository_name}" \
   --lifecycle-policy-text "file://${lifecycle_policy_file}"
+
+  if [[ ${BUILDKITE_PLUGIN_CREATE_ECR_SCAN_ON_PUSH} ]]; then
+    echo '--- Setting ECR image scaning'
```
Comment (72636c) [inline hooks/pre-command]:
```suggestion
    echo '--- Setting ECR image scanning configuration'
```

File: hooks/pre-command [hunk 3]
```diff
@@ -37,6 +37,14 @@ upsert_ecr() {
...
+    aws ecr put-image-scanning-configuration \
+      --repository-name ${repository_name} \
```
Comment (72636c) [inline hooks/pre-command]: Sorry, we should get around to setting up CI with `ShellCheck` someday 😞

```suggestion
      --repository-name "${repository_name}" \
```

File: hooks/pre-command [hunk 4]
```diff
@@ -37,6 +37,14 @@ upsert_ecr() {
...
+      --image-scanning-configuration \
+      scanOnPush=${BUILDKITE_PLUGIN_CREATE_ECR_SCAN_ON_PUSH}
```
Comment (72636c) [inline hooks/pre-command]:
```suggestion
      scanOnPush="${BUILDKITE_PLUGIN_CREATE_ECR_SCAN_ON_PUSH}"
```

File: plugin.yml [hunk 5]
```diff
@@ -9,6 +9,8 @@ configuration:
       type: string
     name:
       type: string
+    scan-on-push:
+      type: string
```
Comment (72636c) [inline plugin.yml]: Should we treat this as a boolean?

```suggestion
      type: boolean
```

Here's an example of matching a YAML _true_: https://github.com/buildkite-plugins/docker-buildkite-plugin/blob/5745e8f7f3cc99343b25bb57381288192df3331c/hooks/command#L90

**Review by 72636c — COMMENTED** [review]
Thanks for the PR 🙏! Just a few comments.

**Review by hari2192 — COMMENTED** [review]
Address comments

File: plugin.yml (hunk 5 above)
Comment (72636c) [reply plugin.yml]: We should update this to boolean:

https://github.com/seek-oss/create-ecr-buildkite-plugin/blob/d16f8bc2c1db595a5afc29f30575aeb945cf7c8c/README.md#L40

And this to check for the many trues in YAML per the link above:

https://github.com/seek-oss/create-ecr-buildkite-plugin/blob/d16f8bc2c1db595a5afc29f30575aeb945cf7c8c/hooks/pre-command#L41

Comment (hari2192) [reply plugin.yml]: Both comments addressed.

File: hooks/pre-command [hunk 6]
```diff
@@ -37,6 +39,14 @@ upsert_ecr() {
   aws ecr put-lifecycle-policy \
   --repository-name "${repository_name}" \
   --lifecycle-policy-text "file://${lifecycle_policy_file}"
+
+  if [[ "${BUILDKITE_PLUGIN_CREATE_ECR_SCAN_ON_PUSH:-}" =~ ^(true|on|1)$ ]]; then
```
Comment (72636c) [inline hooks/pre-command:41]: Ah sorry, I misled you. We'll need to accept falsy values too, and then map back to one of `true` or `false` for the AWS CLI command to work.

If it keeps the code simpler, we could do this with an if else:

```shell
if [[ "${BUILDKITE_PLUGIN_CREATE_ECR_SCAN_ON_PUSH:-}" =~ ^(true|on|1)$ ]]; then
  aws ecr ... scanOnPush=true
elif [[ "${BUILDKITE_PLUGIN_CREATE_ECR_SCAN_ON_PUSH:-}" =~ ^(false|off|0)$ ]]; then
  aws ecr ... scanOnPush=false
fi
```

Comment (hari2192) [reply hooks/pre-command:41]: Agree. I have covered both enabling and disabling of image scan as per your suggestion.

File: hooks/pre-command [hunk 7]
```diff
@@ -14,6 +14,8 @@ ecr_exists() {
 }
 
 upsert_ecr() {
+  printenv
```
Comment (72636c) [inline hooks/pre-command]: We should remove this, as sensitive information can be stored in environment variables

**Review by 72636c — COMMENTED** [review]
Awesome, thanks for your patience! Just one last thing and I'll merge.

File: hooks/pre-command (hunk 7 above)
Comment (hari2192) [reply hooks/pre-command]: My Bad, leftover from testing.... Removed.
Thanks for picking this.

**Review by 72636c — APPROVED** [review]
🎉

**Pull request merged**